CRM/Financeextras/Hook/PageRun/ContributionPageTab.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use CRM_Financeextras_ExtensionUtil as E;
+use CRM_Financeextras_Hook_PageRun_PageRunInterface as PageRunInterface;
+
+class CRM_Financeextras_Hook_PageRun_ContributionPageTab implements PageRunInterface {
+
+  /**
+   * Handles the hook invocation.
+   *
+   * @param CRM_Core_Page $page
+   */
+  public function handle($page) {
+    if (!$this->shouldHandle($page)) {
```

Comment:
no point in having this check given it is already checked inside `financeextras.php`